engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/WorkflowExecutionActor.scala

```diff
@@ -430,6 +433,29 @@ case class WorkflowExecutionActor(params: WorkflowExecutionActorParams)
     )
   }
 
+   /*
+    * If this ExpressionKey has an AnonymousExpressionNode that feeds a task call input, use the backend IoFunctionSet
+    * instead of the engine's IoFunctionSet to properly handle `writeFile` or `readFile` invocations.
+    */
+  private def processRunnableTaskCallInputExpression(key: ExpressionKey, data: WorkflowExecutionActorData, anonymousExpressionNode: AnonymousExpressionNode): ErrorOr[WorkflowExecutionDiff] = {
+    import cats.syntax.either._
+    import cats.syntax.option._
+
+    val taskCallNode = anonymousExpressionNode.callNodeReceivingInput.get(())
+
+    (for {
+      callNode <- (Option(taskCallNode) collect { case t: TaskCallNode => t }).toValidNel(s"Not a TaskCallNode: $taskCallNode").toEither
+      jobKey <- data.executionStore.keyForNode(callNode).toValidNel(s"No job key found for call node $callNode").toEither
+      backendJobDescriptorKey <- Option(jobKey).collectFirst({ case c: BackendJobDescriptorKey => c }).toValidNel(s"Job key not a BackendJobDescriptorKey: $jobKey").toEither
+      backendName <- workflowDescriptor
+        .backendAssignments.get(callNode).toValidNel(s"Cannot find an assigned backend for call ${callNode.fullyQualifiedName}").toEither
+      backendInitializationData = params.initializationData.get(backendName)
+      backendFactory <- backendFactories.get(backendName).toValidNel(s"Cannot find a backend factory for backend $backendName").toEither
+      functions = backendFactory.expressionLanguageFunctions(workflowDescriptor.backendDescriptor, backendJobDescriptorKey, backendInitializationData)
+      diff <- key.processRunnable(functions, data.valueStore, self).toEither
+    } yield diff).toValidated
```

Comment:
Huge ToL: I could be way off here, but I'm curious if someday cromwell could use _a lot less_ conversion back-and-forth between `Try`/`Either`/`Coproduct`/`Validated`/etc.

Reply:
Yeah I'm not super happy with the way it is now, and it's not a great excuse but this follows the pattern elsewhere in this file. In the short term I'll attempt to DRY and sugar a bit but I very much agree with what you're saying.